compiler/p/codegen/OMRRegisterDependency.cpp

```diff
@@ -844,74 +844,86 @@ void TR_PPCRegisterDependencyGroup::assignRegisters(TR::Instruction   *currentIn
       {
       map.addDependency(_dependencies[i], i);
 
-      virtReg = _dependencies[i].getRegister();
       dependentRegNum = _dependencies[i].getRealRegister();
 
       if (dependentRegNum != TR::RealRegister::SpilledReg)
          {
-         if (virtReg->getKind() == TR_GPR)
-            num_gprs++;
-         else if (virtReg->getKind() == TR_FPR)
-            num_fprs++;
-         else if (virtReg->getKind() == TR_VRF)
-            num_vrfs++;
+         virtReg = _dependencies[i].getRegister();
+         switch (virtReg->getKind())
+            {
+            case TR_GPR:
+               ++numGPRs;
+               break;
+            case TR_FPR:
+               ++numFPRs;
+               break;
+            case TR_VRF:
+               ++numVRFs;
+               break;
+            case TR_CCR:
+               if (!virtReg->getAssignedRegister())
```

Comment:
> Is this condition enough? In an OOL section if a virtual register is not assigned to a real register it may not always mean the register is spilled.

That's true, it's not accurate, and `haveSpilledCCRs` should really be `mayHaveSpilledCCRs` if the check stays as-is, but a false positive is not harmful here because we can always assign CCRs before GPRs without penalty (see my response to @zl-wang).

The most accurate/least costly check I can think of given the current RA implementation would be `getAssignedRegister() != NULL && getBackingStorage() != NULL`.

> Small nitpick as well: `getAssignedRegister` does not return a boolean, so shouldn't the if be checking for `virtReg->getAssignedRegister() == NULL` (I really wish we supported `nullptr`).

Personal preference but yes you are right.